File: src/network/packet/impl/ServerSyncPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;
use pocketcloud\cloud\server\CloudServer;

final class ServerSyncPacket extends CloudPacket implements ClientboundPacket {

    public function __construct(
        private readonly ?CloudServer $server = null,
        private readonly bool $removal = false
    ) {}

    public function handle(ServerClient $client): void {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->server, $this->removal);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function getServer(): ?CloudServer {
        return $this->server;
    }

    public function isRemoval(): bool {
        return $this->removal;
    }

    public static function create(CloudServer $server, bool $removal): self {
        return new self($server, $removal);
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/impl/ServerSyncPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.cloud.network.packet.ClientboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.cloud.network.packet.data.PacketData;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class ServerSyncPacket extends CloudPacket implements ClientboundPacket {

    private CloudServer server;
    private boolean removal;

    public ServerSyncPacket(CloudServer server, boolean removal) {
        this.server = server;
        this.removal = removal;
    }

    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(server, removal);
    }

    @Override
    public void decodePayload(PacketData packetData) {}

    public static ServerSyncPacket create(CloudServer server, boolean removal) {
        return new ServerSyncPacket(server, removal);
    }
}
